homeassistant/components/enigma2/media_player.py

```diff
@@ -0,0 +1,196 @@
+"""Support for Enigma2 media players."""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.media_player import MediaPlayerDevice
+from homeassistant.helpers.config_validation import (PLATFORM_SCHEMA)
+from homeassistant.components.media_player.const import (
+    SUPPORT_NEXT_TRACK, SUPPORT_PAUSE, SUPPORT_PREVIOUS_TRACK, SUPPORT_TURN_ON,
+    SUPPORT_TURN_OFF, SUPPORT_VOLUME_MUTE, SUPPORT_VOLUME_SET,
+    MEDIA_TYPE_TVSHOW)
+from homeassistant.const import (
+    CONF_HOST, CONF_NAME, STATE_OFF, STATE_ON, CONF_PORT)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['openwebif.py==0.9']
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'Enigma2 Media Player'
+DEFAULT_PORT = 80
+DEFAULT_TIMEOUT = 0
+
+SUPPORTED_ENIGMA2 = SUPPORT_VOLUME_SET | SUPPORT_VOLUME_MUTE | \
+                          SUPPORT_TURN_OFF | SUPPORT_NEXT_TRACK | \
+                          SUPPORT_PREVIOUS_TRACK | \
+                          SUPPORT_TURN_ON | SUPPORT_PAUSE
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+})
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up of an enigma2 media player."""
+    name = config[CONF_NAME]
+
+    # Generate a configuration for the Enigma2 library
+    remote_config = {
+        'name': 'HomeAssistant',
+        'description': config[CONF_NAME],
+        'port': config[CONF_PORT],
+        'host': config[CONF_HOST],
+        'timeout': DEFAULT_TIMEOUT,
+    }
+
+    add_devices([Enigma2Device(name, remote_config)], True)
+
+
+class Enigma2Device(MediaPlayerDevice):
+    """Representation of an Enigma2 box."""
+
+    def __init__(self, name, config):
+        """Initialize the Enigma2 device."""
+        import openwebif.api
+        self._state = None
+        self._name = name
+        self.e2_box = openwebif.api.CreateDevice(config['host'],
+                                                 port=config['port'])
+
+        self.volume = 10
+        self.current_service_channel_name = None
+        self.current_programme_name = None
+        self.current_service_ref = None
+        self.muted = False
+        self.picon_url = None
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self._state
+
+    @property
+    def supported_features(self):
+        """Flag of media commands that are supported."""
+        return SUPPORTED_ENIGMA2
+
+    def turn_off(self):
+        """Turn off media player."""
+        if self.state == STATE_ON:
```

Comment:
We should not do these checks, that's up to the caller of those functions to decide.